src/Symfony/Component/Encryption/EncryptionInterface.php

```diff
@@ -0,0 +1,137 @@
+     * encryption should be used when you are not sending the encrypted message to
+     * anyone else.
+     *
+     * Example: You store a value on disk or in a cookie and don't want anyone else
+     * to read it.
+     *
+     * Symmetric encryption is in theory weaker than asymmetric encryption.
+     *
+     * <code>
+     *     $key = $encryption->generateKey();
+     *     $ciphertext = $encryption->encrypt('input', $key);
+     *     $message = $encryption->decrypt($ciphertext, $key);
+     * </code>
+     *
+     * @param string       $message plain text version of the message
+     * @param KeyInterface $myKey   a key that holds a string secret
+     *
+     * @return string the output
+     *
+     * @throws EncryptionException
+     */
+    public function encrypt(string $message, KeyInterface $myKey): string;
+
+    /**
+     * Get an encrypted version of the message that only the recipient can read.
+     *
+     * Asymmetric encryption uses a "key pair" ie a public key and a private key.
+     * It is safe to share your public key, but the private key should always be
+     * kept a secret.
+     *
+     * When Alice and Bob wants to communicate they share their public keys with
+     * each other. Alice will encrypt a message with bobs public key. When Bob
+     * receive the message, he will decrypt it with his private key.
+     *
+     *
+     * <code>
+     *     $bobKey = $encryption->generateKey();
+     *     $bobPublicKey = serialize($bobKey->createPublicKey());
+     *
+     *     // Bob sends the public key to Alice
+     *
+     *     $key = unserialize($bobPublicKey);
+     *     $ciphertext = $encryption->encryptFor('input', $key);
+     *
+     *     // Alice sends ciphertext to bob
+     *
+     *     $message = $encryption->decrypt($ciphertext, $bobKey);
+     * </code>
+     *
+     * @param string       $message      plain text version of the message
+     * @param KeyInterface $recipientKey A key with a public key of the recipient
+     *
+     * @return string the output
+     *
+     * @throws EncryptionException
+     */
+    public function encryptFor(string $message, KeyInterface $recipientKey): string;
+
+    /**
+     * Get an encrypted version of the message that only the recipient can read.
+     * The recipient can also verify who sent the message.
+     *
+     * Asymmetric encryption uses a "key pair" ie a public key and a private key.
+     * It is safe to share your public key, but the private key should always be
+     * kept a secret.
+     *
+     * When Alice and Bob wants to communicate they share their public keys with
+     * each other. Alice will encrypt a message with keypair [ alice_private, bob_public ].
+     * When Bob receive the message, he will decrypt it with keypair [ bob_private, alice_public ].
+     *
+     * <code>
+     *     $aliceKey = $encryption->generateKey();
+     *     $bobKey = $encryption->generateKey();
+     *     $keypair = $aliceKey->createKeypair($bobKey);
```

Comment:
This looks wrong to me. When Alice generates the keypair for signing, using `$bobKey` should not be possible. Alice only knows Bob's public key, not bob's private key.

Reply:
It is correct. But I also see you point that it is confusing. 
I'll make it longer for clarity.